src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextMiddleware.cs

```diff
@@ -51,8 +57,24 @@ public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicom
 
             dicomRequestContextAccessor.RequestContext = dicomRequestContext;
 
-            // Call the next delegate/middleware in the pipeline
-            await _next(context);
+            ByteCountingStream byteCountingStream = _responseLogStreamFactory.CreateByteCountingResponseLogStream(context.Response.Body);
```

Comment:
Do we need to dispose this?

Reply:
I couldn't find the actual documentation, but I found at least one example where this is done with a `using` - https://weblog.west-wind.com/posts/2020/Mar/29/Content-Injection-with-Response-Rewriting-in-ASPNET-Core-3x#aspnet-core-3x-response-rewriting-with-ihttpresponsestreamfeature

Reply:
Ah thanks. I was worried that disposing it would dispose the response before it was meant to be disposed. Updated.